### PR TITLE
Kevin yoo/biomarkers mockup page

### PR DESCRIPTION
**Pull request by kevinyooky**
This is the setup I will use for the biomarkers for both genes and proteins page. I will continue building these pages out with the other components as I move along.

### CODE REVIEW

File: context/app/static/js/components/Routes/Routes.jsx [hunk 1]
```diff
@@ -271,6 +273,30 @@ function Routes({ flaskData }) {
     throw Error('Intentional client-side-error');
   }
 
+  // if (urlPath.startsWith('/organ/')) {
+  //   return (
+  //     <Route>
+  //       <Organ organ={organ} />
+  //     </Route>
+  //   );
+  // }
```
Comment (john-conroy) [inline context/app/static/js/components/Routes/Routes.jsx:282]: Why is there an additional commented out organ route?

File: CHANGELOG-add-publication-link 2.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- Add link to publications landing page in 'Other' dropdown menu.
```
Comment (john-conroy) [inline CHANGELOG-add-publication-link 2.md:1]: There are duplicated changelog entries because your branch was created from main before the release was made. These will need to be removed if merging main into your branch doesn't do that. Ultimately only your changelog entry should remain.

File: context/app/static/js/components/detailPage/Genes/SummaryBody/SummaryBody.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        />
+      )} */}
+      {/* <Flex>
+        {published_timestamp ? (
+          <StyledCreationDate label="Publication Date" timestamp={published_timestamp} />
+        ) : (
+          <StyledCreationDate label="Creation Date" timestamp={created_timestamp} />
+        )}
+        <StyledModificationDate label="Modification Date" timestamp={last_modified_timestamp} />
+      </Flex> */}
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/Genes/SummaryBody/SummaryBody.jsx:48]: Preference for not leaving commented out, unrelated code.

File: context/app/static/js/components/detailPage/Genes/SummaryBody/SummaryBody.stories.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/Genes/SummaryBody/SummaryBody.stories.js:1]: This story doesn't seem to reflect the Biomarker mockup?

File: context/app/static/js/components/detailPage/Genes/SummaryBody/style.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+const Flex = styled.div`
+  display: flex;
+`;
+
+const StyledCreationDate = styled(LabelledSectionDate)`
+  flex-grow: 1;
+`;
+
+const StyledModificationDate = styled(LabelledSectionDate)`
+  flex-grow: 3;
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/Genes/SummaryBody/style.js:18]: Only what you import and use in `genes/SummaryBody` should be created here.

File: context/app/static/js/pages/Proteins/Proteins.jsx
Comment (john-conroy) [inline context/app/static/js/pages/Proteins/Proteins.jsx:1]: Empty file but is imported from in `Proteins/index.js`?

**Review by john-conroy — CHANGES_REQUESTED** [review]
I think the intent here is to stub out the biomarker pages and to chain PRs from here, which is great, but stubs shouldn't have unused code and this many commented out blocks. The stubs can be very minimal and not necessarily reflect the mockup, but they shouldn't reflect other pages and have duplicate, unused code that could end up being forgotten about. Maybe we can discuss this over a call?

File: context/app/static/js/components/Routes/Routes.jsx (hunk 1 above)
Comment (kevinyooky) [reply context/app/static/js/components/Routes/Routes.jsx:281]: resolved during our meeting

File: CHANGELOG-add-publication-link 2.md (hunk 2 above)
Comment (kevinyooky) [reply CHANGELOG-add-publication-link 2.md]: resolved during our meeting

File: context/app/static/js/components/detailPage/Genes/SummaryBody/SummaryBody.jsx (hunk 3 above)
Comment (kevinyooky) [reply context/app/static/js/components/detailPage/Genes/SummaryBody/SummaryBody.jsx]: resolved

File: context/app/static/js/components/detailPage/Genes/SummaryBody/style.js (hunk 5 above)
Comment (kevinyooky) [reply context/app/static/js/components/detailPage/Genes/SummaryBody/style.js]: resolved

File: context/app/static/js/pages/Proteins/Proteins.jsx
Comment (kevinyooky) [reply context/app/static/js/pages/Proteins/Proteins.jsx:1]: removed

File: context/app/static/js/components/Routes/Routes.jsx [hunk 6]
```diff
@@ -270,6 +271,14 @@ function Routes({ flaskData }) {
     throw Error('Intentional client-side-error');
   }
 
+  if (urlPath.startsWith('/genes/')) {
+    return (
+      <Route>
+        <Genes organ={organ} collection={entity} />
```
Comment (john-conroy) [inline context/app/static/js/components/Routes/Routes.jsx:277]:
```suggestion
        <Genes />
```
You updated the flask route to no longer pass the organ in flask data. I imagine collection/entity is also a vestige of the copied route. You also want to past the gene name in, but that could be done in a later PR if that's your preference.

File: context/app/static/js/components/genes/SummaryBody/SummaryBody.jsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import SummaryPaper from 'js/shared-styles/sections/SectionPaper';
+import LabelledSectionText from 'js/shared-styles/sections/LabelledSectionText';
+
+function SummaryBody() {
+  return (
+    <SummaryPaper>
+      <LabelledSectionText label="Description" bottomSpacing={1}>
+        Our best estimates show there are over 7 billion people on the planet and 300 billion stars in the Milky Way
```
Comment (john-conroy) [inline context/app/static/js/components/genes/SummaryBody/SummaryBody.jsx:9]: Add a comment to make it more clear this is temporary?

File: context/app/static/js/pages/Genes/Genes.jsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  const assaysId = 'Assays';
+  const samplesId = 'Samples';
+
+  const shouldDisplaySection = {
+    [summaryId]: true,
+    [hraId]: true,
+    [referenceId]: true,
+    [assaysId]: true,
+    [samplesId]: true,
+  };
```
Comment (john-conroy) [inline context/app/static/js/pages/Genes/Genes.jsx:21]:
```suggestion
function Genes() {
  const summaryId = 'Summary';

  const shouldDisplaySection = {
    [summaryId]: true,
  };
```

Comment (john-conroy) [reply context/app/static/js/pages/Genes/Genes.jsx:21]: You only have a summary section. Might need to reformat the suggestion.

**Review by john-conroy — CHANGES_REQUESTED** [review]
Thanks Kevin! A lot cleaner.

File: context/app/static/js/pages/Genes/Genes.jsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import { getSections } from 'js/shared-styles/sections/TableOfContents/utils';
+import Summary from 'js/components/genes/Summary';
+import { FlexRow, Content } from './style';
+
+function Genes() {
+  const summaryId = 'Summary';
+  const hraId = 'Human Reference Atlas';
+  const referenceId = 'Reference-Based Analysis';
+  const assaysId = 'Assays';
+  const samplesId = 'Samples';
```
Comment (NickAkhmetov) [inline context/app/static/js/pages/Genes/Genes.jsx:13]: These constants are being reinstantiated on every render and should be moved outside of the component:

```suggestion
const summaryId = 'Summary';
const hraId = 'Human Reference Atlas';
const referenceId = 'Reference-Based Analysis';
const assaysId = 'Assays';
const samplesId = 'Samples';

function Genes() {
```

**Review by NickAkhmetov — COMMENTED** [review]
Great job! I only have a few small suggestions.

File: context/app/static/js/pages/Genes/Genes.jsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    [hraId]: true,
+    [referenceId]: true,
+    [assaysId]: true,
+    [samplesId]: true,
+  };
+
+  const sectionOrder = Object.entries(shouldDisplaySection)
+    .filter(([, shouldDisplay]) => shouldDisplay)
+    .map(([sectionName]) => sectionName);
+  const sections = new Map(getSections(sectionOrder));
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/pages/Genes/Genes.jsx:26]: I think we should lean towards only memoizing expensive calculations.

[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/pages/Genes/Genes.jsx:26]: I don't think adding complexity to the code is worth the minimal performance benefit if any. There may be some differences with react18, but I mostly agree with https://kentcdodds.com/blog/usememo-and-usecallback#so-when-should-i-usememo-and-usecallback.

[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/pages/Genes/Genes.jsx:26]: https://www.joshwcomeau.com/react/usememo-and-usecallback/ is informative as well.

File: context/app/static/js/components/Routes/Routes.jsx (hunk 6 above)
Comment (kevinyooky) [reply context/app/static/js/components/Routes/Routes.jsx:277]: Completely missed that part!

File: context/app/static/js/components/genes/SummaryBody/SummaryBody.jsx (hunk 7 above)
Comment (kevinyooky) [reply context/app/static/js/components/genes/SummaryBody/SummaryBody.jsx:9]: Done

**Review by NickAkhmetov — APPROVED** [review]
🎈🎉
